testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergHiveViewsCompatibility.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.iceberg;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.tempto.ProductTest;
+import io.trino.tempto.assertions.QueryAssert;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tests.product.TestGroups.ICEBERG;
+import static io.trino.tests.product.TestGroups.STORAGE_FORMATS;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestIcebergHiveViewsCompatibility
+        extends ProductTest
+{
+    @Test(groups = {ICEBERG, STORAGE_FORMATS})
+    public void testIcebergHiveViewsCompatibility()
+    {
+        try {
+            // ensure clean env
+            cleanup();
+
+            List<QueryAssert.Row> hivePreexistingTables = onTrino().executeQuery("SHOW TABLES FROM hive.default").rows().stream()
+                    .map(list -> row(list.toArray()))
+                    .collect(toList());
+            List<QueryAssert.Row> icebergPreexistingTables = onTrino().executeQuery("SHOW TABLES FROM iceberg.default").rows().stream()
+                    .map(list -> row(list.toArray()))
+                    .collect(toList());
+
+            onTrino().executeQuery("CREATE TABLE hive.default.hive_table AS SELECT 1 bee");
+            onTrino().executeQuery("CREATE TABLE iceberg.default.iceberg_table AS SELECT 2 snow");
+
+            onTrino().executeQuery("USE hive.default"); // for sake of unqualified table references
+            onTrino().executeQuery("CREATE VIEW hive.default.hive_view_qualified_hive AS SELECT * FROM hive.default.hive_table");
+            onTrino().executeQuery("CREATE VIEW hive.default.hive_view_unqualified_hive AS SELECT * FROM hive_table");
+            onTrino().executeQuery("CREATE VIEW hive.default.hive_view_qualified_iceberg AS SELECT * FROM iceberg.default.iceberg_table");
+            // this should probably fail but it does not now; testing current behavior as a documentation
```

Comment:
may be also add similar with `USE iceberg.default` i.e. `SELECT * FROM hive.default.iceberg_table`?

Reply:
qualified names should work the same way, so this isn't much different from `CREATE VIEW iceberg.default.iceberg_view_qualified_hive AS SELECT * FROM hive.default.hive_table` ?

Reply:
I meant in context of your previous comment, where enabling redirection would change the behavior.  (Didn't mean to emphasize on `USE`, that was confusing.) 

my point was testing an iceberg view that refers to a qualified iceberg table from hive catalog (i.e. `hive.default.iceberg_table`), that'd be read ultimately through iceberg when redirection is enabled.  (similar to how it'd impact the below query)

Reply:
> I meant in context of your previous comment, where enabling redirection would change the behavior.

That's impossible to cover here, since redirections will be opt-in feature. 

i think these tests will need to be revisited in https://github.com/trinodb/trino/pull/8340.
We may want to have product test coverage for both situations: redirects not enabled, redirects enabled.